service/common/src/main/java/org/apache/polaris/service/config/ReservedProperties.java

```diff
@@ -0,0 +1,127 @@
+  default boolean shouldThrow() {
+    return true;
+  }
+
+  /**
+   * Removes reserved properties from a planned change to an entity. If `shouldThrow`returns true,
+   * this will throw an IllegalArgumentException.
+   *
+   * @param existingProperties The properties currently present for an entity
+   * @param updateProperties The properties present in an update to an entity
+   * @return The keys from the new key list which are not reserved properties
+   */
+  default Map<String, String> removeReservedPropertiesFromUpdate(
+      Map<String, String> existingProperties, Map<String, String> updateProperties)
+      throws IllegalArgumentException {
+    Map<String, String> updatePropertiesWithoutReservedProperties =
+        removeReservedProperties(updateProperties);
+    for (var entry : updateProperties.entrySet()) {
+      // If a key was removed from the update, we substitute in the existing value as to not remove
+      // it
+      if (!updatePropertiesWithoutReservedProperties.containsKey(entry.getKey())) {
+        if (existingProperties.containsKey(entry.getKey())) {
+          updatePropertiesWithoutReservedProperties.put(
+              entry.getKey(), existingProperties.get(entry.getKey()));
+        }
+      }
+    }
+    return updatePropertiesWithoutReservedProperties;
+  }
+
+  /**
+   * Removes reserved properties from a list of input property keys. If `shouldThrow`returns true,
+   * this will throw an IllegalArgumentException.
+   *
+   * @param properties A map of properties to remove reserved properties from
+   * @return The keys from the input list which are not reserved properties
+   */
+  default Map<String, String> removeReservedProperties(Map<String, String> properties)
+      throws IllegalArgumentException {
+    Map<String, String> results = new HashMap<>();
+    List<String> prefixes = reservedPrefixes();
+    for (var entry : properties.entrySet()) {
+      boolean isReserved = false;
+      for (String prefix : prefixes) {
+        if (entry.getKey().startsWith(prefix)) {
+          String message =
+              String.format("Property '%s' matches reserved prefix '%s'", entry.getKey(), prefix);
+          if (shouldThrow()) {
+            throw new IllegalArgumentException(message);
+          } else {
+            LOGGER.debug(message);
+            isReserved = true;
+          }
+        }
+      }
+      if (!isReserved) {
+        results.put(entry.getKey(), entry.getValue());
+      }
+    }
+    return results;
+  }
+
+  /** See {@link #removeReservedProperties(Map)} */
+  default List<String> removeReservedProperties(List<String> properties)
+      throws IllegalArgumentException {
+    Map<String, String> propertyMap =
+        properties.stream().collect(Collectors.toMap(k -> k, k -> ""));
+    Map<String, String> filteredMap = removeReservedProperties(propertyMap);
+    return filteredMap.keySet().stream().toList();
```

Comment:
If we had an `isReserved(name)` method, we could simply filter the original list.

Reply:
It's a good idea, but I think we'd still want to delegate to the other `removeReservedProperties` implementation so that we don't have to duplicate the logic to log/throw

Reply:
good point